src/Utilities/Compiler/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -132,9 +132,21 @@ public static bool DerivesFrom([NotNullWhen(returnValue: true)] this ITypeSymbol
             INamedTypeSymbol? iDisposable,
             INamedTypeSymbol? iAsyncDisposable)
         {
-            return type.IsReferenceType &&
-                (IsInterfaceOrImplementsInterface(type, iDisposable) ||
-                 IsInterfaceOrImplementsInterface(type, iAsyncDisposable));
+            if (type.IsReferenceType)
+            {
+                return IsInterfaceOrImplementsInterface(type, iDisposable)
+                    || IsInterfaceOrImplementsInterface(type, iAsyncDisposable);
+            }
+
+#if NET_ANALYZERS
+            if (type.IsRefLikeType)
```

Comment:
Please update the doc comment accordingly.

Reply:
I can add the description of this specific case to the doc comment (e.g. something like _or is a ref-like type with a non-private void Dispose() method_), but I feel like if it would be better if the doc comment spoke about applicability of `using` statement rather than listed all the specific cases it is looking for. E.g. I'd replace the whole comment with _Indicates if the given type is disposable (can be used in a `using` or `await using` statement)._ What do you think?

Reply:
Sure, that sounds a reasonable change to me.